src/test/perfTest.node.ts

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { spawn } from 'child_process';
+import glob from 'glob';
+import * as path from '../platform/vscode-path/path';
+import { EXTENSION_ROOT_DIR_FOR_TESTS } from './constants.node';
+
+class TestRunner {
+    public async start() {
+        console.log('Start Test Runner');
+        const testFiles = await new Promise<string[]>((resolve, reject) => {
+            glob(`**/*perf.test.js`, (error, files) => {
+                if (error) {
+                    return reject(error);
+                }
+                resolve(files);
+            });
+        });
+
+        // warm up with a basic notebook operation before running the tests
+        await this.launchTest('notebookCellExecution.perf.test', true);
```

Comment:
One thing that I might be missing on this. Per earlier comments and the comments in the tracker it looked like each test is measuring a full test run, including activation. If so what's the warmup needed for?

Reply:
I figured opening vscode for the first time after downloading and installing might have perf implications, and adding more tests would mean that whichever test goes first gets the perf hit.